codecov.gemspec

```
@@ -12,7 +12,7 @@ Gem::Specification.new do |s|
   s.homepage              = 'https://github.com/codecov/codecov-ruby'
   s.license               = 'MIT'
   s.platform              = Gem::Platform::RUBY
-  s.required_ruby_version = '~> 2.4'
+  s.required_ruby_version = '>= 2.4', '< 3.1'
```

Why do you avoid `3.x` versions? They have to be compatible with `3.0`, I see no reason to fear and change it later with every new minor version.

```suggestion
  s.required_ruby_version = '>= 2.4', '< 4'
```

---

Based on the other closed PR I thought we would need to be more conservative, but your suggestion seems fine to me.

---

> Based on the other closed PR I thought we would need to be more conservative

I think, any of extremes is bad. 😅 We should stick reasonable middle, I think.

---

Ruby doesn't follow semantic versioning

---

> Ruby doesn't follow semantic versioning, at least not like a gem since it's not a gem...

Any proofs? Like why, or what type of versioning does it follow? Semver is not only about gems, it's about any software (MRI is software), and you can stick it or not. And I haven't noticed another type of versioning for Ruby, like Chrome has or RubyMine (based more on calendar).

---

> Like why, or what type of versioning does it follow?

ruby 3.0 was named 2.8 in start of it's development cycle, so I  think it released as 3.0 only as Matz wanted to promote ruby x3 mantra

I'm not sure if there is a lot of breaking changes in ruby 3.0, only for keyword arguments, but I'm pretty sure there was some breaking changes for some updates before, like if you update from 2.0 to 2.7 (just an example, not sure), which is defenetly not semver friendly

So I think ruby just it's own version scheme, it's try to mimic semver, but not 100% compatible with it

---

Oh, I found a link below (why did you write multiple messages about the same?). Quote:

> https://www.ruby-lang.org/en/news/2013/12/21/ruby-version-policy-changes-with-2-1-0/

So, yes, this is the information about Ruby may have incompatible things in minor versions, so it's not completely Semver and we can be "more conservative" here with something like `< 3.1`.

Again, `>= 2.4, < 4` is better than just `>= 2.4`, but I can agree here and now that `< 3.1` is even better for Ruby.

Sorry for my mistakes and ignorance.